src/routing/mod.rs

```diff
@@ -36,10 +35,22 @@ mod or;
 
 pub use self::{method_filter::MethodFilter, or::Or};
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+struct RouteId(u64);
+
+impl RouteId {
+    fn next() -> Self {
+        use std::sync::atomic::{AtomicU64, Ordering};
+        static ID: AtomicU64 = AtomicU64::new(0);
+        Self(ID.fetch_add(1, Ordering::SeqCst))
+    }
+}
+
 /// The router type for composing handlers and services.
-#[derive(Debug, Clone)]
+#[derive(Clone)]
 pub struct Router<S> {
```

Comment:
I am surprised this is still a generic type. So even though route matching is no longer done through the "type chain" of routes, that type chain still exists?

Reply:
The type chain still exists since we need to store the services/handlers somewhere. They are stored together with a unique id and that same id is stored in matchit's `Node` type. When a request comes in we match it with the `Node` which returns an id if a route matches. We then call nested chain of services which compares the id with the service and if it matches it'll call the service otherwise they'll call the next service in the chain which repeats the same process.

Reply:
Is OpenAPI schema generation still the main reason you wouldn't want to store type-erased routes in the nodes instead? Or am I making that sound easier than it actually is?

Reply:
Thats the still the main reason. Would also have a small perf impact but I'm not too worried about that.